lamindb/models/_from_values.py

```diff
@@ -345,50 +339,84 @@ def _bulk_create_dicts_from_df(
+    """Check if the field is an id field."""
+    # id field is a unique field that's not a relation
+    field = field.field
+    if field.unique and not field.is_relation:
+        return True
+    return False
+
+
 def _get_organism_record(  # type: ignore
-    field: StrField, organism: str | Record, force: bool = False
-) -> Record:
+    field: FieldAttr,
+    organism: str | Record | None = None,
+    values: Iterable = [],
+    using_key: str | None = None,
+) -> Record | None:
     """Get organism record.
 
     Args:
         field: the field to get the organism record for
         organism: the organism to get the record for
-        force: whether to force fetching the organism record
+        values: the values to get the organism record for
+        using_key: the db to get the organism record for
+
+    Returns:
+        The organism record if:
+            The organism FK is required for the registry
+            The field is not unique or the organism is not None
     """
-    registry = field.field.model  # type: ignore
-    check = True
-    if not force and hasattr(registry, "_ontology_id_field"):
-        check = field.field.name != registry._ontology_id_field  # type: ignore
-        # e.g. bionty.CellMarker has "name" as _ontology_id_field
-        if not registry._ontology_id_field.endswith("id"):
-            check = True
-
-    if _has_organism_field(registry) and check:
-        from bionty._bionty import create_or_get_organism_record
+    registry = field.field.model
+    field_str = field.field.name
+    check = not _is_simple_field_unique(field=field) or organism is not None
+
+    if field_str == "ensembl_gene_id" and len(values) > 0 and organism is None:  # type: ignore
+        return _organism_from_ensembl_id(values[0], using_key)  # type: ignore
 
-        if field and not isinstance(field, str):
-            field = field.field.name
+    if _require_organism(registry) and check:
+        from bionty._bionty import create_or_get_organism_record
 
         organism_record = create_or_get_organism_record(
-            organism=organism, registry=registry, field=field
+            organism=organism, registry=registry, field=field_str
         )
         if organism_record is not None:
-            return organism_record
+            return organism_record.save()
+
+
+def _organism_from_ensembl_id(id: str, using_key: str | None) -> Record | None:  # type: ignore
+    import bionty as bt
 
+    from .artifact import Artifact  # has to be here to avoid circular imports
 
-def _ensembl_prefix(id: str, field: StrField, organism: Record | None) -> str | None:
-    if field.field.name == "ensembl_gene_id" and organism is None:  # type: ignore
-        if id.startswith("ENSG"):
-            organism = "human"  # type: ignore
-        elif id.startswith("ENSMUSG"):
-            organism = "mouse"  # type: ignore
+    ensembl_prefixes = (
+        Artifact.using("laminlabs/bionty-assets")
```

Comment:
How often is this called? Latency won't be great still but of course we should improve latency so much that it can be called in such situtations. ☺️ 

So, I think it's good that you're using it.

Reply:
Only if the database has never seen an ensembl id prefix. This won't be called, if ensembl ids with the same prefix are already in the DB.